cgsmiles/read_cgsmiles.py

```diff
@@ -146,28 +146,54 @@ def read_cgsmiles(pattern):
 
         # here we check if the atom is followed by a cycle marker
         # in this case we have an open cycle and close it
-        for token in pattern[stop:]:
-            # we close a cycle
-            if token.isdigit() and token in cycle:
-                cycle_edges.append((current, cycle[token]))
-                del cycle[token]
-            # we open a cycle
-            elif token.isdigit():
-                cycle[token] = current
-            # we found a ring indicator
-            elif token == "%":
-                ring_marker = _get_percent(pattern, stop)
-                # we close the ring
+        ring_marker = ""
+        multi_ring = False
+        ring_bond_order = default_bond_order
+        for rdx, token in enumerate(pattern[stop:]):
+            if multi_ring and not token.isdigit():
                 if ring_marker in cycle:
-                    cycle_edges.append((current, cycle[ring_marker]))
+                    cycle_edges.append((current,
+                                        cycle[ring_marker][0],
+                                        cycle[ring_marker][1]))
                     del cycle[ring_marker]
-                    break
-                # we open a new ring
-                cycle[_get_percent(pattern, stop)] = current
-                break
+                else:
+                    cycle[ring_marker] = [current, ring_bond_order]
+                multi_ring = False
+                ring_marker = ""
+                ring_bond_order = default_bond_order
+
+            # we open a new multi ring
+            if token == "%":
+                multi_ring = True
+                ring_marker = '%'
+            # we open a ring or close
+            elif token.isdigit():
+                ring_marker += token
+                if not multi_ring:
+                    # we have a single digit marker and it is in
+                    # cycle so we close it
+                    if ring_marker in cycle:
+                        cycle_edges.append((current,
+                                            cycle[ring_marker][0],
+                                            cycle[ring_marker][1]))
+                        del cycle[ring_marker]
+                    # the marker is not in cycle so we update cycles
+                    else:
+                        cycle[ring_marker] = [current, ring_bond_order]
+                    ring_marker = ""
+                    ring_bond_order = default_bond_order
+            # we found bond_order
+            elif token in symbol_to_order:
+                ring_bond_order = symbol_to_order[token]
```

Comment:
Higher bond orders between normal beads is still not allowed, right? *E.g.* `[#C]=[#B]1.[#A]1`. I'm afraid this will result in a double bond between B and A

Reply:
No, it is allowed and even encouraged. It completely replaces the ring index stuff. Ultimately, there are use cases where the ring syntax becomes ridiculous. Sterols are one example but also THC. So what previously was `[#C]1[#B]1` now should be `[#C]=[#B]`

Reply:
Yes, agree with all you say, but I'm suspecting there's a bug in the parser here

Reply:
no this one raises a SyntaxError

Reply:
That also doesn't sound right, does it? My concern is that *any* specified "normal" bond order sticks around and gets used by the next ring bond

Reply:
Yes because you apply ring indices to consecutive atoms; ahh that's your concern let me see

Reply:
they get cleared in 195. Consider `[#A]=[#B]1[#C][#D]1`. When we parse atom #B `prev_bond_order=2` and `bond_order=1` as B is not followed by anything.

Reply:
I assume this is resolved?

Reply:
Aaah, yes.
Excellent.